wallet/wallet.go

```diff
@@ -303,27 +303,71 @@ func (w *Wallet) SetChainSynced(synced bool) {
 	w.chainClientSyncMtx.Unlock()
 }
 
+// shouldOnlyScanDefaultKeyScopes is a helper function that allows the wallet to
+// determine whether it should only scan the chain for addresses within its
+// default scopes. The reason for this method existing is due to a wallet bug,
+// that has now been addressed, in which change addresses could potentially be
+// created within a non-default key scope if it shared an address schema with a
+// default one. In order to ensure older wallets reflect their proper balance,
+// we gate whether they should scan for all existing key scopes by their
+// birthday height.
+func (w *Wallet) shouldOnlyScanDefaultKeyScopes(birthday *waddrmgr.BlockStamp) bool {
+	const (
+		// Wallets with a birthday after this height are not susceptible
+		// to the bug described, so they can safely only scan the
+		// default key scopes.
+		mainNetHeight = 623000
```

Comment:
How were these heights obtained? The heights when the prior active addr change landed in master?

Reply:
Instead, I think we should just have this be based off of a database migration version which detects if a change address was ever created under a non-default scope. If so, then we need to check for everything constantly, otherwise we can only start to watch the default scopes from there.